clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -2700,18 +2694,31 @@ protected void processStartOfIncrementalPush(
       ControlMessage startOfIncrementalPush,
       PartitionConsumptionState partitionConsumptionState) {
     CharSequence startVersion = ((StartOfIncrementalPush) startOfIncrementalPush.controlMessageUnion).version;
-    ingestionNotificationDispatcher
-        .reportStartOfIncrementalPushReceived(partitionConsumptionState, startVersion.toString());
+    if (!batchReportIncPushStatusEnabled || partitionConsumptionState.isComplete()) {
+      ingestionNotificationDispatcher
+          .reportStartOfIncrementalPushReceived(partitionConsumptionState, startVersion.toString());
+    }
   }
 
   protected void processEndOfIncrementalPush(
       ControlMessage endOfIncrementalPush,
       PartitionConsumptionState partitionConsumptionState) {
-    // TODO: it is possible that we could turn incremental store to be read-only when incremental push is done
     CharSequence endVersion = ((EndOfIncrementalPush) endOfIncrementalPush.controlMessageUnion).version;
-    // Reset incremental push version
-    ingestionNotificationDispatcher
-        .reportEndOfIncrementalPushReceived(partitionConsumptionState, endVersion.toString());
+    LOGGER.info(
+        "DEBUGGING BATCH REPORT ENABLED? {}, PCS completed {}",
+        batchReportIncPushStatusEnabled,
+        partitionConsumptionState.isComplete());
+    if (!batchReportIncPushStatusEnabled || partitionConsumptionState.isComplete()) {
+      ingestionNotificationDispatcher
+          .reportEndOfIncrementalPushReceived(partitionConsumptionState, endVersion.toString());
+    } else {
+      // TODO: We will still need to update push status store, for now it is still writing entry by entry.
```

Comment:
I think for now it might be OK. Just it is not optimal to my understanding, but I think it should not impact the backend for now as volume has been reduced.